docs/FAQ.md

```diff
@@ -318,3 +318,166 @@ kubectl delete ns calico-apiserver calico-system
+
+  ```
+  * provider.aws: version = "~> 2.10"
+  * provider.local: version = "~> 1.2"
+  * provider.null: version = "~> 2.1"
+  * provider.random: version = "~> 2.1"
+  * provider.template: version = "~> 2.1"
+
+  Terraform has been successfully initialized!
+  ```
+
+#### plan
+
+The `terraform plan` command reads terraform state and configuration files and displays a list of changes and actions that need to be performed to bring the state in line with the configuration. It's a convenient way to test changes before applying them. When used with the `-out` parameter, it saves a batch of changes to a specified file that can later be used with `terraform apply`. Call example:
+
+  ```bash
+  $ terraform plan
+  # ~600 rows skipped
+  Plan: 82 to add, 0 to change, 0 to destroy.
+
+  ------------------------------------------------------------------------
+
+  Note: You didn't specify an "-out" parameter to save this plan, so Terraform
+  can't guarantee that exactly these actions will be performed if
+  "terraform apply" is subsequently run.
+  ```
+
+#### apply
+
+The `terraform apply` command scans `.tf` in the current directory and brings the state to the configuration described in them by making changes in the infrastructure. By default, `plan` with a continuation dialog is performed before applying. Optionally, you can specify a saved plan file as input:
+
+  ```bash
+  $ terraform apply
+  # ~600 rows skipped
+  Plan: 82 to add, 0 to change, 0 to destroy.
+
+  Do you want to perform these actions?
+    Terraform will perform the actions described above.
+    Only 'yes' will be accepted to approve.
+
+    Enter a value: yes
+
+  Apply complete! Resources: 82 added, 0 changed, 0 destroyed.
+  ```
+
+We do not always need to re-read and compare the entire state if small changes have been added that do not affect the entire infrastructure. For this, you can use targeted `apply`; for example:
+
+  ```bash
+  $ terraform apply -target helm_release.kibana
+  ```
+
+Details can be found [here](https://www.terraform.io/docs/cli/run/index.html)
+
+> The first time, the `apply` command must be executed in the layers in order: first layer1, then layer2. Infrastructure `destroy` should be done in the reverse order.
+
+## Update terraform version
+
+Change terraform version in this files
+
+`terraform/.terraform-version` - the main terraform version for tfenv tool
+
+`.github/workflows/terraform-ci.yml` - the terraform version for github actions need for `terraform-validate` and `terraform-format`.
+
+Terraform version in each layer.
+```
+terraform/layer1-aws/main.tf
+terraform/layer2-k8s/main.tf
+```
+
+## Update terraform providers
+
+Change terraform providers version in this files
+
+```
+terraform/layer1-aws/main.tf
+terraform/layer2-k8s/main.tf
+```
+
+When we changed terraform provider versions, we need to update terraform state. For update terraform state in layers we need to run this command:
+
+```
+terragrunt run-all init -upgrade
```

Comment:
The text above is related to terraform, but here terragrunt is used. Is it better to provide commands for both ways?

Reply:
fixed